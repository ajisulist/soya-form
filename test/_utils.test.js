import {
  isArray,
  isField,
  removeObject,
  updateArray,
  removeArray,
  moveArray,
  getFieldNames,
  getDisplayName,
  createValidate,
  createValidateAsync,
  createGetName,
} from '../src/_utils';

describe('utils', () => {
  it('isArray', () => {
    expect(isArray([])).toBe(true);
    expect(isArray({})).toBe(false);
    expect(isArray(undefined)).toBe(false);
    expect(isArray(null)).toBe(false);
    expect(isArray(1)).toBe(false);
    expect(isArray('')).toBe(false);
  });

  it('isField', () => {
    expect(isField({
      errorMessages: [],
      isEnabled: true,
      isValidating: false,
      touched: false,
      value: null,
    })).toBe(true);
    expect(isField([])).toBe(false);
    expect(isArray({})).toBe(false);
    expect(isArray(undefined)).toBe(false);
    expect(isArray(null)).toBe(false);
    expect(isArray(1)).toBe(false);
    expect(isArray('')).toBe(false);
  });

  it('removeObject', () => {
    expect(removeObject({
      a: 1,
      b: 2,
    }, 'a')).toMatchSnapshot();
  });

  it('updateArray', () => {
    expect(updateArray([1, 2, 4], [3], -1)).toMatchSnapshot();
    expect(updateArray([1, 2, 4], [3], 2)).toMatchSnapshot();
    expect(updateArray([1, 2, 3], [4], 3)).toMatchSnapshot();
  });

  it('removeArray', () => {
    expect(removeArray([1, 2, 4], -1)).toMatchSnapshot();
    expect(removeArray([1, 2, 4], 2)).toMatchSnapshot();
    expect(removeArray([1, 2, 3], 3)).toMatchSnapshot();
  });

  it('moveArray', () => {
    expect(moveArray([1, 2, 4], 0, 2)).toMatchSnapshot();
    expect(moveArray([1, 2, 4], 3, -1)).toMatchSnapshot();
    expect(moveArray([1, 2, 3], -1, 4)).toMatchSnapshot();
  });

  it('getFieldNames', () => {
    expect(getFieldNames([])).toMatchSnapshot();
    expect(getFieldNames('foo')).toMatchSnapshot();
    expect(getFieldNames(['foo', 'bar'])).toMatchSnapshot();
  });

  it('getDisplayName', () => {
    expect(getDisplayName('Prefix', () => {})).toMatchSnapshot();
  });

  it('createValidate', () => {
    expect(createValidate([
      () => 'Error 1',
      () => ({}),
      () => 1,
      () => null,
    ])()).toMatchSnapshot();
  });

  it('createValidateAsync', async () => {
    expect(await createValidateAsync([
      () => Promise.resolve('Error 1'),
      () => Promise.resolve({}),
      () => Promise.resolve(1),
      () => Promise.resolve(null),
    ])()).toMatchSnapshot();
  });

  it('createGetName', () => {
    expect(createGetName(['foo', 'bar'], 0)()).toMatchSnapshot();
    expect(createGetName(['foo', 'bar'], 0)(0)).toMatchSnapshot();
  })
});
