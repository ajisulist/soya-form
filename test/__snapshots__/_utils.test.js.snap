// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils createGetName 1`] = `
Array [
  "foo",
  "bar",
  0,
]
`;

exports[`utils createGetName 2`] = `
Array [
  "foo",
  "bar",
  0,
  0,
]
`;

exports[`utils createValidate 1`] = `
Array [
  "Error 1",
  Object {},
]
`;

exports[`utils createValidateAsync 1`] = `
Array [
  "Error 1",
  Object {},
]
`;

exports[`utils getDisplayName 1`] = `"Prefix(Component)"`;

exports[`utils getFieldNames 1`] = `Array []`;

exports[`utils getFieldNames 2`] = `
Array [
  "foo",
]
`;

exports[`utils getFieldNames 3`] = `
Array [
  "foo",
  "bar",
]
`;

exports[`utils moveArray 1`] = `
Array [
  2,
  4,
  1,
]
`;

exports[`utils moveArray 2`] = `
Array [
  1,
  2,
  4,
]
`;

exports[`utils moveArray 3`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`utils removeArray 1`] = `
Array [
  1,
  2,
  4,
]
`;

exports[`utils removeArray 2`] = `
Array [
  1,
  2,
]
`;

exports[`utils removeArray 3`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`utils removeObject 1`] = `
Object {
  "b": 2,
}
`;

exports[`utils updateArray 1`] = `
Array [
  1,
  2,
  4,
]
`;

exports[`utils updateArray 2`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`utils updateArray 3`] = `
Array [
  1,
  2,
  3,
  4,
]
`;
